File: tests/test_workon_cmd.py
from tests.utils import CLITestCase
from task_forge.sdk.types import Task
from task_forge.cli.workon_cmd import run


class TestWorkon(CLITestCase):
    def test_workon_cmd(self):
        task = Task(id="test", title="test", priority=1)
        self.client.tasks.current.return_value = Task(title="test", priority=100)
        self.client.tasks.get.return_value = task
        args = self.args()
        args["<ID>"] = task.id
        run(args)
        self.client.tasks.get.assert_called_with(task.id)
        self.assertEqual(task.priority, 101)

File: tests/test-work-on-cmd.ts
import {
    cli,
    listTasks,
    generateTask,
    generateUser,
    getContextName
} from './utils';

describe('task work-on', () => {
    test('given an id makes that task the highest priority task', async () => {
        const { token } = await generateUser();
        const { id } = await generateTask(token);
        await generateTask(token, { priority: 2 });
        await cli(`work-on ${id}`, token);
        const tasks = await listTasks(token);
        for (const task of tasks) {
            if (task.id === id) {
                expect(task.priority).toBe(3);
            } else {
                expect(task.priority).toBe(2);
            }
        }
    });
});
